From ea8464f8f07db3d497f34349411e92d26ed293be Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Tue, 29 Apr 2014 17:51:09 +0800
Subject: [PATCH 4/4] ENGR00310049 - the camera APP cannot open anymore after
 record video and panorama

by switch back/front camera. 80%

media_profile.xml set to 640x480, but TVIN not support 640x480.
Modify profile resolution if only one preview resolution and not same.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 src/com/android/camera/VideoModule.java | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 61cffe3..deb8f72 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -596,6 +596,20 @@ public class VideoModule implements CameraModule,
         // TODO: This should be checked instead directly +1000.
         if (mCaptureTimeLapse) quality += 1000;
         mProfile = CamcorderProfile.get(mCameraId, quality);
+
+		//For TVIN, either 720x576 or 720x480, media_profile can't give a suitable resolution,
+		//so change the profile value.
+		List<Size> sizes = mParameters.getSupportedPreviewSizes();
+		if( (sizes.size() == 1 ) &&
+			( (sizes.get(0).width != mProfile.videoFrameWidth) || (sizes.get(0).height != mProfile.videoFrameHeight) ) ) {
+			Log.w(TAG, "Only support one preview resolution diff with profile, change profile frome " +
+				mProfile.videoFrameWidth + "x" + mProfile.videoFrameHeight + " to " +
+				sizes.get(0).width + "x" + sizes.get(0).height);
+
+			mProfile.videoFrameWidth = sizes.get(0).width;
+			mProfile.videoFrameHeight = sizes.get(0).height;
+		}
+
         getDesiredPreviewSize();
         mPreferenceRead = true;
     }
-- 
1.8.0

